server/src/main/java/com/genymobile/scrcpy/wrappers/WindowManager.java

```diff
@@ -198,4 +210,64 @@ public void unregisterDisplayWindowListener(IDisplayWindowListener listener) {
             Ln.e("Could not unregister display window listener", e);
         }
     }
+
+    @TargetApi(AndroidVersions.API_29_ANDROID_10)
+    private Method getGetDisplayImePolicyMethod() throws NoSuchMethodException {
+        if (getDisplayImePolicyMethod == null) {
+            try {
+                getDisplayImePolicyMethod = manager.getClass().getMethod("getDisplayImePolicy", int.class);
+            } catch (NoSuchMethodException e) {
+                getDisplayImePolicyMethod = manager.getClass().getMethod("shouldShowIme", int.class);
+                getDisplayImePolicyMethodNewVersion = false;
+            }
+        }
+        return getDisplayImePolicyMethod;
+    }
+
+    @TargetApi(AndroidVersions.API_29_ANDROID_10)
+    public int getDisplayImePolicy(int displayId) {
+        try {
+            Method method = getGetDisplayImePolicyMethod();
+            if (getDisplayImePolicyMethodNewVersion) {
+                    return (int) method.invoke(manager, displayId);
+            } else {
+                boolean shouldShow = (boolean) method.invoke(manager, displayId);
+                return shouldShow ? DISPLAY_IME_POLICY_LOCAL : DISPLAY_IME_POLICY_FALLBACK_DISPLAY;
+            }
+        } catch (ReflectiveOperationException e) {
+            Ln.e("Could not invoke method", e);
+            return DISPLAY_IME_POLICY_FALLBACK_DISPLAY;
+        }
+    }
+
+    @TargetApi(AndroidVersions.API_29_ANDROID_10)
+    private Method getSetDisplayImePolicyMethod() throws NoSuchMethodException {
+        if (setDisplayImePolicyMethod == null) {
+            try {
+                setDisplayImePolicyMethod = manager.getClass().getMethod("setDisplayImePolicy", int.class, int.class);
+            } catch (NoSuchMethodException e) {
```

Comment:
The change between `shouldShowIme()` and `getDisplayImePolicy()` was performed by [this commit](https://android.googlesource.com/platform/frameworks/base/+/acd7a80462a17111d04d2a5eeab154f33f0bfa54%5E%21/) introduced in Android 12, so it can be checked directly with `Build.VERSION.SDK_INT < AndroidVersions.API_31_ANDROID_12`.

Reply:
Thank you.